azure-core.gemspec

```diff
@@ -26,12 +26,12 @@ Gem::Specification.new do |s|
   s.homepage    = 'http://github.com/Azure/azure-ruby-asm-core'
   s.license     = 'Apache License, Version 2.0'
   s.files       = `git ls-files`.split("\n").reject { |f|  f.start_with?("test/unit") }
-  s.extensions  = 'ext/mkrf_conf.rb'
```

Comment:
You are absolutely right. Now that we rely on our guidance in `Readme.md` for user to install correct version of Nokogiri via Rubygems, and resolved Bundler version issue, we can remove that. I have updated the PR for the purpose.